spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/SqlRowWriter.scala

```diff
@@ -23,9 +26,16 @@ class SqlRowWriter(val table: TableDef, val selectedColumns: IndexedSeq[ColumnRe
     require(row.size == columnNames.size, s"Invalid row size: ${row.size} instead of ${columnNames.size}.")
     for (i <- 0 until row.size) {
       val colValue = row(i)
-      buffer(i) = converters(i).convert(colValue)
+      buffer(i) = Try(converters(i).convert(colValue)) match {
+        case Success(value) => value
+        case Failure(ex: IllegalArgumentException) =>
```

Comment:
The most exceptions I've seen that were hard to track down were either `IllegalArgumentException` or its subclass.